mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb

```diff
@@ -2830,3 +2830,75 @@ resource "google_container_node_pool" "np2" {
   }
 `, cluster, np1, np2)
 }
+
+func testAccContainerNodePool_withDefaultTaintSuppress(cluster, np string, add_custom_taint bool) string {
+	custom_taint := ""
+	if add_custom_taint {
+		custom_taint = `
+taint {
+	key = "custom_taint"
+	value = "test"
+	effect = "NO_SCHEDULE"
+}`
+	}
+	return fmt.Sprintf(`
+data "google_container_engine_versions" "central1a" {
+  location = "us-central1-a"
+}
+
+resource "google_container_cluster" "cluster" {
+  name               = "%s"
+  location           = "us-central1-a"
+  initial_node_count = 1
+  min_master_version = data.google_container_engine_versions.central1a.latest_master_version
+}
+
+resource "google_container_node_pool" "with_default_gpu_taint" {
+  name               = "%s"
+  location           = "us-central1-a"
+  cluster            = google_container_cluster.cluster.name
+  initial_node_count = 1
+  node_config {
+    machine_type = "n1-standard-1"
+    image_type = "COS_CONTAINERD"
+
+    taint {
+      key    = "nvidia.com/gpu"
+      value  = "present"
+      effect = "NO_SCHEDULE"
+    }
+
+	%s
+
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/logging.write",
+      "https://www.googleapis.com/auth/monitoring",
+    ]
+  }
+}
+`, cluster, np, custom_taint)
+}
+
+func TestAccContainerNodePool_withDefaultTaintSuppress(t *testing.T) {
```

Comment:
We don't need this test; however, we should have exhaustive unit tests for the diff suppress function. See https://github.com/hashicorp/terraform-provider-google/blob/main/google/common_diff_suppress_test.go for examples of what unit tests might look like.